File: ruby/spec/parser/escaped_copy.spec.rb
input = <<~DOC.strip
template:
`key`<template
``k`ey`` < template
```ke``y```     < template
    `` `key` ``     < template
DOC

describe Enolib::Parser do
  describe 'Escaped copies' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end

File: php/spec/tokenizer/escaped_copy.spec.php
<?php declare(strict_types=1);

require_once(__DIR__ . '/util.php');

describe('Escaped copy tokenization', function() {
  given('input', function() {
    return <<<DOC
`key` < template
``ke`y`` < template
```k``ey```    < template
    `` `key` ``    < template
DOC;
  });

  it('works as specified', function() {
    $instructions = inspectTokenization($this->input);

    expect($instructions)->toMatchSnapshot('spec/tokenizer/snapshots/escaped_copy.snap.json');
  });
});
